src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -326,6 +326,7 @@ final case class ProcessApiServiceImpl(
       catalogItem    <- catalogManagementService.getEServiceById(eServiceUuid)
       _              <- assertRequesterAllowed(catalogItem.producerId)(organizationId)
       descriptor     <- assertDescriptorExists(catalogItem, descriptorUuid)
+      _              <- isDraftDescriptor(descriptor)
```

Comment:
This implementation LGTM but I think is not exhaustive.
Here in the process, we manage only the metadata of the file: the file is uploaded on S3 in the BFF.
I suggest to implement this control in both services (tenant process and BFF).

What do you think @galales and @nttdata-rtorsoli ?

Reply:
I see your point and it is actually true for every file upload in the platform.
Honestly I'm worried about duplicating logics in the BFF.
Just for the sake of discussion, another suboptimal solution could be changing the flow in
1. BFF invokes catalog process to add metadata
2. BFF uploads the file
2a. in case of upload failure, BFF invokes catalog process to delete metadata

Reply:
Ok, I'm adding a specific task for this.
Thank you!